[CI] ShardFollowTaskReplicationTests testRetryBulkShardOperations fails with shard closed

This is the first occurrence from what I can tell:

```
org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests > testRetryBulkShardOperations FAILED	
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=79, name=elasticsearch[org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests][generic][T#1], state=RUNNABLE, group=TGRP-ShardFollowTaskReplicationTests]	
        at __randomizedtesting.SeedInfo.seed([60531DCA12EF97EF:2636282B8F900408]:0)	
	
        Caused by:	
        java.lang.AssertionError: java.util.concurrent.ExecutionException: [test/uuid][[test][0]] IndexShardClosedException[CurrentState[CLOSED] Closed]	
            at __randomizedtesting.SeedInfo.seed([60531DCA12EF97EF]:0)	
            at org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests$CcrAction.performOnPrimary(ShardFollowTaskReplicationTests.java:681)	
            at org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests$CcrAction.performOnPrimary(ShardFollowTaskReplicationTests.java:665)	
            at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase$ReplicationAction$PrimaryRef.perform(ESIndexLevelReplicationTestCase.java:645)	
            at org.elasticsearch.action.support.replication.ReplicationOperation.execute(ReplicationOperation.java:109)	
            at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase$ReplicationAction.execute(ESIndexLevelReplicationTestCase.java:612)	
            at org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests$4.lambda$innerSendBulkShardOperationsRequest$0(ShardFollowTaskReplicationTests.java:547)	
	
            Caused by:	
            java.util.concurrent.ExecutionException: [test/uuid][[test][0]] IndexShardClosedException[CurrentState[CLOSED] Closed]	
                at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.getValue(BaseFuture.java:266)	
                at org.elasticsearch.common.util.concurrent.BaseFuture$Sync.get(BaseFuture.java:253)	
                at org.elasticsearch.common.util.concurrent.BaseFuture.get(BaseFuture.java:87)	
                at org.elasticsearch.xpack.ccr.action.ShardFollowTaskReplicationTests$CcrAction.performOnPrimary(ShardFollowTaskReplicationTests.java:676)	
                ... 5 more	
	
                Caused by:	
                [test/uuid][[test][0]] IndexShardClosedException[CurrentState[CLOSED] Closed]	
                    at org.elasticsearch.index.shard.IndexShardOperationPermits.acquire(IndexShardOperationPermits.java:243)	
                    at org.elasticsearch.index.shard.IndexShardOperationPermits.lambda$releaseDelayedOperations$1(IndexShardOperationPermits.java:206)	
                    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:633)	
                    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)	
                    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)	
                    at java.lang.Thread.run(Thread.java:748)
```

Build scan:
https://gradle-enterprise.elastic.co/s/7g3perhddnqe4

--- a/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowTaskReplicationTests.java
+++ b/x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowTaskReplicationTests.java
@@ -126,6 +126,7 @@ public class ShardFollowTaskReplicationTests extends ESIndexLevelReplicationTest
                         equalTo(leaderGroup.getPrimary().getLastKnownGlobalCheckpoint()));
                     followerGroup.assertAllEqual(indexedDocIds.size() - deleteDocIds.size());
                 });
+                assertNull(shardFollowTask.getStatus().getFatalException());
                 shardFollowTask.markAsCompleted();
                 assertConsistentHistoryBetweenLeaderAndFollower(leaderGroup, followerGroup, true);
             }
@@ -170,6 +171,7 @@ public class ShardFollowTaskReplicationTests extends ESIndexLevelReplicationTest
                 assertThat(shardFollowTask.getFailure(), nullValue());
                 int expectedDoc = docCount;
                 assertBusy(() -> followerGroup.assertAllEqual(expectedDoc));
+                assertNull(shardFollowTask.getStatus().getFatalException());
                 shardFollowTask.markAsCompleted();
                 assertConsistentHistoryBetweenLeaderAndFollower(leaderGroup, followerGroup, hasPromotion == false);
             }
@@ -325,6 +327,7 @@ public class ShardFollowTaskReplicationTests extends ESIndexLevelReplicationTest
                             equalTo(leadingPrimary.getLastKnownGlobalCheckpoint()));
                         assertConsistentHistoryBetweenLeaderAndFollower(leaderGroup, followerGroup, true);
                     });
+                    assertNull(shardFollowTask.getStatus().getFatalException());
                 } finally {
                     shardFollowTask.markAsCompleted();
                 }
@@ -397,6 +400,7 @@ public class ShardFollowTaskReplicationTests extends ESIndexLevelReplicationTest
                     follower.recoverReplica(follower.addReplica());
                 }
                 assertBusy(() -> assertConsistentHistoryBetweenLeaderAndFollower(leader, follower, false));
+                assertNull(followTask.getStatus().getFatalException());
                 followTask.markAsCompleted();
             }
         }
@@ -675,7 +679,7 @@ public class ShardFollowTaskReplicationTests extends ESIndexLevelReplicationTest
                     request.getOperations(), request.getMaxSeqNoOfUpdatesOrDeletes(), primary, logger);
                 TransportWriteActionTestHelper.performPostWriteActions(primary, request, ccrResult.location, logger);
             } catch (InterruptedException | ExecutionException | IOException e) {
-                throw new AssertionError(e);
+                throw new RuntimeException(e);
             }
             listener.onResponse(new PrimaryResult(ccrResult.replicaRequest(), ccrResult.finalResponseIfSuccessful));
         }

